### PR TITLE
Robustify detection if `bokeh-mathjax.js` bundle is needed

### LINKED ISSUE #13607: [BUG] Multiple inline math elements in different axes causing axis labels to disappear
### Software versions

Python version        :  3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:37:07) [Clang 15.0.7 ]
IPython version       :  (not installed)
Tornado version       :  6.3.2
Bokeh version         :  3.3.2
BokehJS static path   :  /opt/homebrew/Caskroom/mambaforge/base/envs/short_period_variable_neural_network_paper/lib/python3.11/site-packages/bokeh/server/static
node.js version       :  v20.5.0
npm version           :  9.8.0
jupyter_bokeh version :  (not installed)
Operating system      :  macOS-14.2-arm64-arm-64bit

### Expected behavior

Multiple inline math elements should be rendered regardless of what the opposing axis in the figure contains.

### Observed behavior

In some configurations, when multiple inline math elements are included in both axes of a figure, both axis labels are not rendered.

### Example code

In this example, neither axis label is rendered.

```Python
from bokeh.io import show
from bokeh.plotting import figure as Figure

figure0 = Figure(x_axis_label=r'$$a$$\[b\]', y_axis_label=r'$$c$$\[d\]', width=200, height=200)
figure0.line(x=[0, 1], y=[0, 1])
show(figure0)
```
image0

If the second inline math element is removed from either axis label, both axis labels are correctly rendered. That is if either `\[b\]` or `\[d\]` is removed, both labels are correctly rendered.

Notably, if the above figure with incorrect rendering is placed in a `Row` with a duplicate figure that had either `\[b\]` or `\[d\]` removed, then all 4 axis labels are correctly rendered. For example, the following code has a `Row` containing the failing figure above, but when placed in this `Row`, the figure correctly renders.

```Python
from bokeh.io import show
from bokeh.models import Row
from bokeh.plotting import figure as Figure

figure1 = Figure(x_axis_label=r'$$a$$\[b\]', y_axis_label=r'$$c$$', width=200, height=200)
figure1.line(x=[0, 1], y=[0, 1])
figure0 = Figure(x_axis_label=r'$$a$$\[b\]', y_axis_label=r'$$c$$\[d\]', width=200, height=200)
figure0.line(x=[0, 1], y=[0, 1])
row = Row(figure1, figure0)
show(row)
```
image1

### PR DESCRIPTION
**Pull request by mattpap**
Also simplifies relevant unit tests and improves types in `bokeh.embed.bundle`.

fixes #13607

### CODE REVIEW

File: tests/unit/bokeh/embed/test_bundle.py [hunk 1]
```diff
@@ -38,61 +38,57 @@
...
-    from bokeh.models import PlainText
-    test_plaintext = PlainText("$$sin(x)$$")
-    return test_plaintext
+    glplot = figure(output_backend="webgl")
+    glplot.scatter([1, 2], [2, 3])
+    return glplot
+
+def table() -> models.DataTable:
+    table = models.DataTable()
+    return table
```
Comment (bryevdv) [inline tests/unit/bokeh/embed/test_bundle.py]: I have a vague recollection that the fixtures imports were there to defer the `bokeh.models` import for some reason, but I don't recall any detail, and maybe it's not relevant any longer, if it ever was. 

In any case, as long as we are simplifying (here and below):

```suggestion
    return models.DataTable()
```

**Review by bryevdv — APPROVED** [review]
LGTM one suggestion
